src/charm.py

```diff
@@ -341,48 +338,32 @@ def _config_changed(self, event):
         # First execute the setup script in 2 conditions:
         # - First run (when no services are planned in pebble)
         # - Change in important S3 parameter (comparing value with envVars in pebble plan)
-        script = f"{SCRIPT_PATH}/pod_setup"
         if (
             self._is_config_valid()
             and self.model.unit.is_leader()
             and self._should_run_setup(current_plan, previous_s3_info)
         ):
-
+            script = f"{SCRIPT_PATH}/pod_setup"
             logger.debug("Executing setup script (%s)", script)
             process = container.exec(
                 [script], environment=self._create_discourse_environment_settings()
             )
-            try:
-                self.model.unit.status = MaintenanceStatus(f"Executing setup {script}")
-                stdout, _ = process.wait_output()
-                logger.debug("%s stdout: %s", script, stdout)
-            except ExecError as e:
-                logger.error("%s command exited with code %d. Stderr:", script, e.exit_code)
-                for line in e.stderr.splitlines():
-                    logger.error("    %s", line)
-                logger.debug("%s stdout: %s", script, e.stdout)
-                self.model.unit.status = BlockedStatus(f"Error while executing {script}")
-                raise
```

Comment:
I has been very very helpful to have the logs of the script when working with S3, I also used it a lot when debugging the integration tests on Github. I don't think it's a good idea to remove this part, unless you know a better way to get the script output

Reply:
I plan to. This is still a draft and not even properly tested